### PR TITLE
docs: Add code review guide

### LINKED ISSUE #785: [Feature Request] Guideline for PR review
### Required prerequisites

- [X] I have searched the [Issue Tracker](https://github.com/camel-ai/camel/issues) and [Discussions](https://github.com/camel-ai/camel/discussions) that this hasn't already been reported. (+1 or comment there if it has.)
- [ ] Consider asking first in a [Discussion](https://github.com/camel-ai/camel/discussions/new).

### Motivation

Guideline for PR review

### Solution

_No response_

### Alternatives

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Wendong-Fan**
## Description

Add code review guide

## Motivation and Context

 `close #785 ` 

- [x] I have raised an issue to propose this change ([required](https://github.com/camel-ai/camel/issues) for new features and bug fixes)

## Types of changes

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds core functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Documentation (update in the documentation)
- [ ] Example (update in the folder of example)

## Implemented Tasks

- [ ] Subtask 1
- [ ] Subtask 2
- [ ] Subtask 3

## Checklist

Go over all the following points, and put an `x` in all the boxes that apply.
If you are unsure about any of these, don't hesitate to ask. We are here to help!

- [ ] I have read the [CONTRIBUTION](https://github.com/camel-ai/camel/blob/master/CONTRIBUTING.md) guide. (**required**)
- [ ] My change requires a change to the documentation.
- [ ] I have updated the tests accordingly. (*required for a bug fix or a new feature*)
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -42,14 +42,56 @@ Ensuring excellent documentation and thorough testing is absolutely crucial. Her
...
+- Functionality
+  - Correctness: Does the code perform the intended task? Are edge cases handled?
+  - Testing: Is there sufficient test coverage? Do all tests pass?
+  - Security: Are there any security vulnerabilities introduced by the change?
+  - Performance: Does the code introduce any performance regressions?
+
+- Code Quality
+  - Readability: Is the code easy to read and understand? Is it well-commented where necessary?
+  - Maintainability: Is the code structured in a way that makes future changes easy?
+  - Style: Does the code follow the project’s style guidelines?
```
Comment (WHALEEYE) [inline CONTRIBUTING.md:75]: do we have a style guideline for now?

Comment (willshang76) [reply CONTRIBUTING.md:75]: I would suggest the Google style guides.(https://google.github.io/styleguide/pyguide.html) They are actively maintained. If we decide to use them, then all reviewers should get familiar with them. There may be some existing pre-commit tests regarding the styles, which we could leverage.

Comment (Wendong-Fan) [reply CONTRIBUTING.md:75]: For now we will do the style/format check using ruff, I will update the doc to make it more clear

Comment (Wendong-Fan) [reply CONTRIBUTING.md:75]: currently we are using ruff to format the code, I will update the doc to make it more clear

Comment (WHALEEYE) [reply CONTRIBUTING.md:75]: I agree to keep using ruff as a code format and lint checker, but I think their checks are too rough for a PR, and passing the checks of ruff should be the prerequisite for raising a PR. When it comes to the reviewing of PR, we need a more specific standard, so that we can do faster and better, and make sure the codebase of CAMEL is consistent. (Also I think the contributors should also be aware of the code style we are using so maybe we can declare the code style in contribution guide)

Comment (Wendong-Fan) [reply CONTRIBUTING.md:75]: updated code style part

Comment (lightaime) [reply CONTRIBUTING.md:75]: Hi @Wendong-Fan. Our code style is a bit different from https://google.github.io/styleguide/pyguide.html. For example the dosctring is quite different. We may need to write it down with some examples.

Comment (lightaime) [reply CONTRIBUTING.md:75]: Hi @WHALEEYE. This is a good question. We should put the docstring in of `__init__` under the class instead of inside the method since this provide a better way for devlopers to use when they are using code editor like VS code.

More informative:
image

Not informative:
image

There are many inconsistences in the code base. We should change them all to putting the docstring in of `__init__` under the class instead of inside the method. @Wendong-Fan @WHALEEYE

Comment (lightaime) [reply CONTRIBUTING.md:75]: Also we should use r-string and the default value we can write like: (default: :obj:`None`).

Comment (Wendong-Fan) [reply CONTRIBUTING.md:75]: I will create a new page to document a detailed review checklist. Over time, we can add items to this list, transforming it into a comprehensive guideline
